Base/BacnetReadAccessSpecification.cs

```diff
@@ -21,8 +21,7 @@ public static object Parse(string value)
             if (string.IsNullOrEmpty(value)) return ret;
             var tmp = value.Split(':');
             if (tmp.Length < 2) return ret;
-            ret.objectIdentifier.type = (BacnetObjectTypes)Enum.Parse(typeof(BacnetObjectTypes), tmp[0]);
-            ret.objectIdentifier.instance = uint.Parse(tmp[1]);
+            ret.objectIdentifier = new BacnetObjectId((BacnetObjectTypes)Enum.Parse(typeof(BacnetObjectTypes), tmp[0]), uint.Parse(tmp[1]));
```

Comment:
What is the point of changing this (here and in similar places)? I know structs are not something we love in C# but since it's still a struct, it doesn't change anything yet.

Reply:
1) It better be a struct, or we would be in **real** trouble since it is not immutable.
2) I changed it here because I had to change 'type' to 'Type' and 'instance' to 'Instance' anyway, and we really should make BacnetObjectId immutable.

I just checked and realized that this can already be done easily - there are only two places left that need to be changed, so I guess I will just do it :wink:

Reply:
Sounds good. So maybe only public getters for those properties without backing fields?